code/controllers/subsystem/bluespace_exploration.dm

```diff
@@ -0,0 +1,542 @@
+	var/datum/data_holder/bluespace_exploration/data = data_holder
+	var/obj/docking_port/mobile/shuttle = SSshuttle.getShuttle(data.shuttle_id)
+	//Place a dock somewhere in the new place
+	var/sanity = PLACEMENT_TRIES
+	var/docking_failed = TRUE
+	while(sanity > 0)
+		sanity --
+		//Try to find a place for the docking port
+		if(!away_mission_port)
+			away_mission_port = new()
+			away_mission_port.name = "Bluespace Drive Port"
+			away_mission_port.id = "exploration_away"
+			away_mission_port.height = shuttle.height
+			away_mission_port.width = shuttle.width
+			away_mission_port.dheight = shuttle.dheight
+			away_mission_port.dwidth = shuttle.dwidth
+		var/max_size = max(away_mission_port.width, away_mission_port.height)
+		away_mission_port.forceMove(locate(rand(max_size, world.maxx - max_size), rand(max_size, world.maxx - max_size), data_holder.z_value))
+		//Check if blocked
+		var/blocked = FALSE
+		for(var/turf/T as() in away_mission_port.return_turfs())
+			if(T.density)
+				blocked = TRUE
+				break
+		if(blocked)
+			CHECK_TICK
+			continue
+		docking_failed = FALSE
+	if(!docking_failed)
+		shuttle.destination = away_mission_port
+	else
+		shuttle.destination = shuttle.previous
+		message_admins("Bluespace exploration docking failed, returning shuttle to home (Sanity check failed to place shuttle)")
+		log_shuttle("Bluespace exploration docking failed, returning shuttle to home (Sanity check failed to place shuttle)")
+		log_runtime("Bluespace exploration docking failed, returning shuttle to home (Sanity check failed to place shuttle)")
+	shuttle.setTimer(shuttle.ignitionTime)
+	current_system = data_holder.target_star_system
+	for(var/datum/space_level/level as anything in bluespace_systems)
+		if(level.z_value == data_holder.z_value)
+			bluespace_systems[level] = BS_LEVEL_USED
+			break
+	generating = world.time + shuttle.ignitionTime + 10
+	generating_level = -1
+
+/datum/controller/subsystem/bluespace_exploration/proc/generate_z_level(datum/data_holder/bluespace_exploration/data_holder)
+	add_to_wipe_queue(data_holder.z_value, data_holder)
+	check_free_levels()
+
+/datum/controller/subsystem/bluespace_exploration/proc/shuttle_translation(shuttle_id, datum/data_holder/bluespace_exploration/data_holder)
+	var/obj/docking_port/mobile/shuttle = SSshuttle.getShuttle(shuttle_id)
+	if(!shuttle || generating)
+		return FALSE
+	generating = INFINITY
+	if(away_mission_port?.get_docked())
+		away_mission_port.delete_after = TRUE
+		away_mission_port.id = null
+		away_mission_port.name = "Old [away_mission_port.name]"
+		away_mission_port = null
+	//Send the shuttle to the transit level
+	shuttle.destination = null
+	shuttle.mode = SHUTTLE_IGNITING
+	shuttle.setTimer(shuttle.ignitionTime)
+	//Clear the z-level after the shuttle leaves
+	//TODO: Remove usage of timers
+	addtimer(CALLBACK(src, .proc/generate_z_level, data_holder), shuttle.ignitionTime + 50, TIMER_UNIQUE)
```

Comment:
TODO going to be done in this PR or the future?

Reply:
thought I did this

Reply:
actually timers is a good bet here